### PR TITLE
networkd: introduce netdev ipvtap

### PR DESCRIPTION
**Pull request by ssahani**
This patch adds netdev ipvtap  is based on the
IP-VLAN network interface, called ipvtap. An ipvtap device can be created
in the same way as an ipvlan device, using 'kind ipvtap', and then accessed
using the tap user space interface.

### CODE REVIEW

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -115,6 +115,10 @@
           <row><entry><varname>ipvlan</varname></entry>
           <entry>An ipvlan device is a stacked device which receives packets from its underlying device based on IP address filtering.</entry></row>
 
+          <row><entry><varname>ipvlan</varname></entry>
```
Comment (yuwata) [inline man/systemd.netdev.xml]: ipvtap

File: man/systemd.netdev.xml [hunk 2]
```diff
@@ -115,6 +115,10 @@
...
+          <entry>An ipvtap device is a stacked device which receives packets from its underlying device based on IP address filtering and can be accessed
+          using the tap user space interface.</entry></row>
```
Comment (yuwata) [inline man/systemd.netdev.xml]: Please do not wrap the line.

**Comment by ssahani** [comment]
Thanks for the review @yuwata . Updated.

**Comment by yuwata** [comment]
@ssahani I made the following:
- missing `IPVTAP=` in directive.network was added.
- basic test for ipvtap was added.

**Comment by yuwata** [comment]
Ubuntu CIs does not test this, and CentOS CI (Arch in KVM with sanitizers) failure is not related.

**Comment by ssahani** [comment]
thank you @yuwata

**Pull request merged**